### PR TITLE
First Arriving API

### PR DESCRIPTION
**Pull request by larskendall**
## Overview
First Arriving API endpoints can now be accessed at `/api/third-party/first-arriving`. Here are each of the views and their endpoints:

**Incident Analysis**
`/api/third-party/first-arriving/incident-analysis`

**Unit Utilization**
`/api/third-party/first-arriving/unit-utilization`

**Turnout Leaderboard**
`/api/third-party/first-arriving/turnout-leaderboard`

## GitHub Issues
- #425 

## Changes
- Added `/api/third-party` route for grouping endpoints by organization for external projects.
- Added `/api/third-party/first-arriving` route for First Arriving endpoints.
- Added endpoints for each First Arriving view.
- Updated `generateTextualSummaries()` to use snake case keys instead of camel case, and updated UI to use the new key names.
- Added `statistics.js` file on server with unit quintile and busyness calculations. The intention is for those functions to eventually be used instead of calculating the same thing client-side for the Units page, but I thought it should wait for a separate PR.

## Steps to Test

**Incident Analysis**
- Make a GET request to `/api/third-party/first-arriving/incident-analysis?fireDepartmentId={x}`.
- You should receive back the latest 20 incidents for that department with the data required for the Incident Analysis view.

**Unit Utilization**
- Make a GET request to `/api/third-party/first-arriving/unit-utilization?fireDepartmentId={x}&interval={x}`. Supported interval values are `shift`, `day`, `week`, `month`, and `year`.
- You should receive data back that matches your requested interval. You can compare this against the data in the app to verify accuracy.

**Turnout Leaderboard**
- Make a GET request to `/api/third-party/first-arriving/turnout-leaderboard?fireDepartmentId={x}&interval={x}`. Supported interval values are `shift`, `day`, `week`, `month`, `year`, and `all`.
- You should receive data back that matches your requested interval. Returned units are sorted by `avg_turnout_duration.seconds`. Any units with missing or erroneous data (infinity/nan/etc) will automatically be sorted to the end of the list with their values set to `null`.

***Unit Utilization** and **Turnout Leaderboard** results are based off of the current date, so in order to get results using older test data, you may need to change line 274 in `first-arriving.controller.js` to `const now = moment('{YOUR_DATE_HERE}').tz(fireDepartment.timezone)` with whatever date best suits your test data.

### CODE REVIEW

File: server/api/third-party/first-arriving/first-arriving.controller.js [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+      }),
+      analysis: generateAnalysis(source),
+    };
+  });
+
+  res.json({ incidents });
+}
+
+export async function getUnitUtilization(req, res) {
+  const interval = req.query.interval;
```
Comment (chopchop505) [inline server/api/third-party/first-arriving/first-arriving.controller.js]: toLowerCase to be safe

File: server/api/third-party/first-arriving/first-arriving.controller.js [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+        },
+      },
+    });
+  }
+
+  res.json({ apparatus });
+}
+
+export async function getTurnoutLeaderboard(req, res) {
+  const interval = req.query.interval;
```
Comment (chopchop505) [inline server/api/third-party/first-arriving/first-arriving.controller.js]: toLowerCase

Comment (larskendall) [reply server/api/third-party/first-arriving/first-arriving.controller.js]: Good call. Done.

File: server/api/third-party/first-arriving/first-arriving.controller.js (hunk 1 above)
Comment (larskendall) [reply server/api/third-party/first-arriving/first-arriving.controller.js]: Good call. Done.

**Comment by chopchop505** [comment]
@larskendall Take a look at this query I added https://github.com/StatEngine/stat-engine/pull/428/commits/01dda58d7847d48fffd2761e0c96ab9d69b3d507

I think that was what you were looking for?  If thats the case can you update  #L247 to handle the new buckets

**Pull request merged**